src/steamship/agents/llms/replicate.py

```diff
@@ -0,0 +1,79 @@
+from typing import List, Optional
+
+from steamship import Block, PluginInstance, Steamship
+from steamship.agents.schema import LLM
+
+PLUGIN_HANDLE = "replicate-llm"
+DEFAULT_MAX_TOKENS = 256
+
+LLAMA_2_70B = "replicate/llama-2-70b-chat"
+DOLLY_2_12B = "replicate/dolly-v2-12b"
+
+DEFAULT_MODEL = LLAMA_2_70B
+
+
+class Replicate(LLM):
+    """LLM that uses Steamship's Replicate plugin to generate completions.
+
+    The default model is Llama 2 70B.
+
+    See the Replicate plugin source code for more information:
+      https://github.com/steamship-plugins/replicate-llms
+    """
+
+    generator: PluginInstance
+    client: Steamship
+
+    def __init__(
+        self,
+        client,
+        model_name: str = LLAMA_2_70B,
+        temperature: float = 0.4,
+        model_version: Optional[str] = None,
+        default_system_prompt: Optional[str] = None,
+        max_tokens: Optional[int] = DEFAULT_MAX_TOKENS,
+        top_p: Optional[float] = None,
+        stop_sequences: Optional[List[str]] = None,
+        *args,
+        **kwargs
+    ):
+        """Create a new instance.
+
+        Valid model names are presently:
+         - replicate/llama-2-70b-chat
+         - replicate/dolly-v2-12b
+        """
+
+        # TODO: We've got whatever the opposite of an encapsulation leak here, in that this is all strongly typed
+        # and defined in the Replicate plugin, yet we can't refer to it here.
+        #
+        # One side-step approach might be to strongly type LLM configuration in the python-client repository, since
```

Comment:
If configuration moves out of this, I'm thinking `__init__` should just take that.

I also don't know if I would call this a strictly side-step approach rather than a normal pattern, publishing an 'interfaces' lib (or bundling common interfaces in a base lib) is an established flow in many cases.

In the case that those configs are just passed into the `__init__`, it might even be possible to just have one general LLM and you provide a config for the one you want to use, since the code otherwise is pretty generic.